src/clang_parser.cpp

```diff
@@ -222,7 +222,10 @@ static bool getBitfield(CXCursor c, Bitfield &bitfield)
   size_t bitfield_offset = clang_Cursor_getOffsetOfField(c) % 8;
   size_t bitfield_bitwidth = clang_getFieldDeclBitWidth(c);
 
-  bitfield.mask = (1 << bitfield_bitwidth) - 1;
+  if (bitfield_bitwidth == 64)
```

Comment:
Shouldn't this be `>=32` as the warning is:

shift exponent 64 is too large for 32-bit type 'int'

Reply:
of maybe 

```
if (width >= sizeof(size_t)) {
  mask = SSIZE_MAX
}
```

Reply:
> Shouldn't this be >=32 as the warning is:

Isn't bitfield_bitwidth >= 32 possible? something like `struct {long x: 48; long y: 16}`

> of maybe
> 
> ```
> if (width >= sizeof(size_t)) {
>   mask = SSIZE_MAX
> }
> ```

ah, this seems better. will fix.

Reply:
Shouldn't we make bitfield.mask an uint64_t then, to avoid issues on 32bit?

Reply:
It makes sense. I also noticed that both clang and gcc support `__int128`. So I added the support of it.

Reply:
I realized that bpftrace does not support `__int128`. So, change it to use `uint64_t`